tools/pubsys/src/aws/ssm/mod.rs

```diff
@@ -135,6 +139,14 @@ pub(crate) async fn run(args: &Args, ssm_args: &SsmArgs) -> Result<()> {
             .context(error::RenderTemplatesSnafu)?;
     trace!("Generated templated parameters: {:#?}", new_parameters);
 
+    // Ensure `--ssm-parameter-output` is set when using `--dry-run`
+    ensure!(
+        !(ssm_args.dry_run && ssm_args.ssm_parameter_output.is_none()),
+        error::MissingArgumentSnafu {
+            argument: "--ssm-parameter-output is required when using --dry-run".to_string()
+        }
+    );
+
```

Comment:
I think the idiomatic way to do this with the `clap` library is [with `ArgGroup`s](https://docs.rs/clap/latest/clap/_derive/_tutorial/index.html#argument-relations). Mind trying that out instead?